server/lib/device/device.getDeviceFeaturesStates.js

```diff
@@ -0,0 +1,69 @@
+const { Op } = require('sequelize');
+const db = require('../../models');
+const { NotFoundError } = require('../../utils/coreErrors');
+const { Error400 } = require('../../utils/httpErrors');
+
+const DEFAULT_OPTIONS = {
+  skip: 0,
+  order_dir: 'ASC',
+  order_by: 'created_at',
+};
+
+/**
+ * @description Get all features states aggregates.
+ * @param {string} selector - Device selector.
+ * @param {object} options - Options of the query.
+ * @param {string} options.from - Start date in UTC format "yyyy-mm-ddThh:mm:ss:sssZ"
+ * or "yyyy-mm-dd hh:mm:ss:sss" (GMT time).
+ * @param {string} [options.to] - End date in UTC format "yyyy-mm-ddThh:mm:ss:sssZ"
+ * or "yyyy-mm-dd hh:mm:ss:sss" (GMT time).
+ * @param {number} [options.take] - Number of elements to return.
+ * @param {number} [options.skip] - Number of elements to skip.
+ * @param {string} [options.attributes] - Possible values (separated by a comma ',' if several): 'id',
+ * 'device_feature_id', 'value', 'created_at' and 'updated_at'. Leave empty to have all the columns.
+ * @returns {Promise<Array>} - Resolve with an array of data.
+ * @example
+ * device.getDeviceFeaturesStates('test-device', {from: '2022-03-31T00:00:00.000Z', to: '2022-03-31T23:59:59.999Z',
+ * take: 100, skip: 10, attributes: 'id,value,created_at'});
+ */
+async function getDeviceFeaturesStates(selector, options) {
+  const deviceFeature = this.stateManager.get('deviceFeature', selector);
+  if (deviceFeature === null) {
+    throw new NotFoundError('DeviceFeature not found');
+  }
+  if (options.from === undefined) {
+    throw new Error400('Start date missing');
```

Comment:
In the core, we try to avoid using HTTP errors (specific to the API), and instead use core errors. In that case, I would use the "BadParameters" error in the `require('../../utils/coreErrors');`